handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+
+1. Backup your SystemLink Server.
+
+1. Attach a volume to store data captured by `nislmigrate`. This will be referenced as `D:\`.
+
+1. Install `nislmigrate`.
+
+1. Run the command `nislmigrate capture --all --secret <your secret> --dir D:\migration`.
+
+1. Detach `D:\`.
+
+    !!!note
+        At this point in the workflow your original SystemLink Server is in an operable state. Be aware if any new data is created or ingested by SystemLink Server at this time it will not be available to your new server.
+
+1. Provision a new Windows server for SystemLink.
+
+1. Provision a PostGreSQL server or replica set.
+
+1. Install and configure the new version of SystemLink Server that supports PostGreSQL.
+
+    !!!important
+        If you are using the single node configuration for SystemLink, the migration from the local instances of MongoDB to PostGreSQL begins automatically. If you are migrating a multi node configuration where the original SystemLink Server used a dedicated MongoDB server or replica set, you must specify the PostGreSQL instance for SystemLink to use before the MongoDB to PostGreSQL migration can occur. The next step in this workflow assumes you are in this multi node configuration.
+
+1. Configure SystemLink to use the newly created PostGreSQL server or replica set and click **Apply**. Refer to [Connecting to a Remote PostgreSQL Database](https://www.ni.com/documentation/en/systemlink/latest/setup/remote-postgres-databse/) for this step.
+
+1. Install `nislmigrate` on your new SystemLink Server.
+
+1. Attach the `D:\` volume used to capture data from your original SystemLink server.
+
+1. Run the command `nislmigrate restore --all --secret <your secret> --dir D:\migration`
+
+    !!!note
+        After this step SystemLink will migrate your test steps, results, and products from MongoDB to PostGreSQL. Depending on the size of your data set this process may take some time.
+
+1. Verify your new SystemLink Server has all the expected migrated data.
+
+## Upgrading Multi Node Configurations
+
+The following workflow describes upgrading a SystemLink Server instance that has been configured to use a dedicated MongoDB server or replica set, a dedicated PostGreSQL server or replica set, and a dedicated file store such as AWS S3.
+
+1. Backup your SystemLink Server, MongoDB server or replica set, PostGreSQL server or replica set, and file store.
+
+    !!!note
+        If S3 is used as your file store the backup step is not needed since this managed internally by AWS.
+
+1. Attach a volume to store data captured by `nislmigrate`. This will be referenced as `D:\`.
+
+1. Install `nislmigrate`.
+
+1. Run the command `nislmigrate capture --tags --dir D:\migration`.
```

Comment:
What about --systems? We migrate SALT configuration.

Reply:
--tags retrieves both current and historical data. Is it okay to replace the historical data? I guess because we drop the database things should be fine, but maybe another warning about not writing to the server during the migration?

Should we have a work item to add an option to only capture tag current values?

Reply:
For very old version of the repository and system state services, their is file data stored on disk that is required for migration. This data will get copied into MongoDB for 21.x versions. We might need to include those services if we support migration from those versions.

Reply:
I added to the note regarding replacing tag history data. I do think we should add an option argument for tags that only migrates current value - FYI @Christian-Nunnally.

Per the old versions of states and repo service, I believe these are outside our support window for the migration tool at this time. I think we can add this, but I'll wait till we have customers requesting this first.